spec/required_env_fetcher_spec.rb

```diff
@@ -1,7 +1,53 @@
 # frozen_string_literal: true
 
 RSpec.describe RequiredEnvFetcher do
-  it "has a version number" do
-    expect(RequiredEnvFetcher::VERSION).not_to be nil
+  describe ".fetch" do
+    let(:key) { "INTERESTING_THING" }
+
+    context "when the key is set in the environment" do
```

Comment:
Nit: the example description might be slightly clearer if we referred to it as the "skip key" vs just "key". What do you think?

Same comment for line 21.

Another alternative would something like "when missing required values are allowed".

Reply:
This is describing when the key we're trying to fetch is present in the environment variables (and 21 is describing when it's not present in the environment).

22 and 44 describe what happens when we are (and aren't) ok with using the default value.

Reply:
Just reworded things a little to see if it makes things clearer?

Reply:
I may just have read too quickly earlier, but it looks good now, thanks!